### PR TITLE
Allow to extend SimpleViewGenerator by replacing loading methods

### PR DESCRIPTION
**Pull request by paolobazzi**
- SimpleViewGenerator make loading method protected (overridable)
- StdDesignDocumentFactory allow to exchange view generator for
subclasses

### CODE REVIEW

File: org.ektorp/src/main/java/org/ektorp/support/StdDesignDocumentFactory.java [hunk 1]
```diff
@@ -19,8 +19,19 @@
  */
 public class StdDesignDocumentFactory implements DesignDocumentFactory {
 
-    public SimpleViewGenerator viewGenerator = new SimpleViewGenerator();
+    protected SimpleViewGenerator viewGenerator;
```
Comment (YannRobert) [inline org.ektorp/src/main/java/org/ektorp/support/StdDesignDocumentFactory.java]: public field made protected :
Could break things for some users

**Comment by paolobazzi** [comment]
Thanks for your input! You're right, my suggested change changes the behavior of the class using the new getter. I like your suggestion of adding a constructor in order to specifiy another `SimpleViewGenerator` instance.
In fact, the code of `SimpleViewGenerator` could remain without changes, but constructing a new instance of `StdDesignDocumentFactory` then implies, that automatically a new `SimpleViewGenerator` is created, even if the generator is later replaced by another instance. Adding a constructor with `SimpleViewGenerator` as parameter would solve this issue.

**Comment by paolobazzi** [comment]
Thanks! What's the process to have this change merged to a next Ektorp release?

**Comment by paolobazzi** [comment]
Thanks Robert!

**Pull request merged**